Automattic/jetpack
# CRM: Fix export crash when custom field name clashes with linked objects

Fixes https://github.com/Automattic/zero-bs-crm/issues/3477

## Proposed changes:

* This PR fixes a fatal error that occurs when exporting contacts with a custom field that had the same name as a linked field (e.g., 'company').
* The way this works is by adding a suffix to a key when there is a name clash, then before the export process uses custom field keys, it checks for clashes. The former is a manual check that has to be added in every place we may have a name clash.

### Other information:


## Jetpack product discussion
https://github.com/Automattic/zero-bs-crm/issues/3477

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:

Steps to replicate this error:

1. Add a custom text field named 'Company' to contacts
2. Add some information to this new custom field for at least one contact
3. Try to export the contact list ( wp-admin/admin.php?page=zbs-export-tools) - Make sure the custom field is selected

The export process will crash.
With this PR the crash no longer happens.
